src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -146,8 +146,13 @@ export class PageControllerBase {
         if (isPageHeading) {
           labelOrLegend.isPageHeading = isPageHeading
 
+          // Check for optional in label
+          const isOptional = labelOrLegend.text.includes('optional')
+
           if (pageTitle) {
-            labelOrLegend.text = pageTitle
+            labelOrLegend.text = isOptional
+              ? pageTitle + ' (optional)'
+              : pageTitle
```

Comment:
I am not convinced this is best solution but I didn't find any other way to do it. 

@alexluckett , @davidjamesstone  - `heading.html` is not the file used to render from my investigation. govukInput in `textfield.html` renders label, so I made above change. 

This solution works. 

Please let me know if I missed anything or better solution.

Reply:
Demo'd and tested few components with Naz. All looks good.

@alexluckett